OSSM-8520 Document migration of network policies created by 2.x (#88728)
openshift/openshift-docs: migrating/checklists/ossm-migrating-network-policies-assembly.adoc

:_mod-docs-content-type: ASSEMBLY
[id=ossm-migrating-network-policies-assembly]
= Migrating network policies from Service Mesh 2 to Service Mesh 3
include::_attributes/common-attributes.adoc[]
:context: ossm-migrating-network-policies-assembly

toc::[]

> In OpenShift {SMProductName} 2, network policies are created by default when `spec.security.manageNetworkPolicy` is set to `true` in the `ServiceMeshControlPlane` resource. During the migration to {SMProduct} 3, these policies are removed.

Reviewer: `OpenShift` is already included in the product name.
```suggestion
In {SMProductName} 2, network policies are created by default when `spec.security.manageNetworkPolicy` is set to `true` in the `ServiceMeshControlPlane` resource. During the migration to {SMProduct} 3, these policies are removed.
```

> It is recommended to recreate your network policies after you have migrated your deployment and workloads. However, for security reasons, you might need to recreate them now, and then set `spec.security.manageNetworkPolicy` to `false` as outlined in the migration checklists.

Reviewer 2: Possible to avoid making a recommendation here and just explaining the options so they can decide what's best for them?
```suggestion
You can re-create your network policies after migrating your deployment and workloads. If your security policies require you to re-create them earlier, do so and then set the `spec.security.manageNetworkPolicy` field to `false` as outlined in the migration checklists.
```

include::modules/ossm-migrating-network-policies-setup-during-migration.adoc[leveloffset=+1]

> .Next step

Reviewer 2: I know it's just one, but we should use the plural heading, even when there is only one (this is stated explicitly for prereqs [here](https://redhat-documentation.github.io/modular-docs/#prerequisites))
```suggestion
.Next steps
```

> * In {SMProduct} 2, set `spec.security.manageNetworkPolicy=` to `false` in your `ServiceMeshControlPlane` resource, and continue with migration checklists.

Reviewer 2: Is this going to have an xref at some point? If not, it can live in the module

> .Next step
> * In {SMProduct} 2, set `spec.security.manageNetworkPolicy=` to `false` in your `ServiceMeshControlPlane` resource, and continue with migration checklists.

Reviewer 2: ```suggestion
.Next step
* In {SMProduct} 2, set the `spec.security.manageNetworkPolicy` field to `false` in your `ServiceMeshControlPlane` resource, and continue with migration checklists.
```

:_mod-docs-content-type: ASSEMBLY
[id=ossm-migrating-network-policies-assembly]
= Migrating network policies from Service Mesh 2 to Service Mesh 3
include::_attributes/common-attributes.adoc[]
:context: ossm-migrating-network-policies-assembly

toc::[]

> In OpenShift {SMProductName} 2, network policies are created by default when `spec.security.manageNetworkPolicy=` is set to `true` in the `ServiceMeshControlPlane` resource. During migration to {SMProduct} 3, these policies are removed.

Reviewer: nit:
```suggestion
In OpenShift {SMProductName} 2, network policies are created by default when `spec.security.manageNetworkPolicy` is set to `true` in the `ServiceMeshControlPlane` resource. During migration to {SMProduct} 3, these policies are removed.
```

> It is recommended to recreate your network policies after you have migrated your deployment and workloads. However, for security reasons, you may need to recreate them now, and then set `spec.security.manageNetworkPolicy=` to `false` as outlined in the migration checklists.

Reviewer: nit:
```suggestion
It is recommended to recreate your network policies after you have migrated your deployment and workloads. However, for security reasons, you may need to recreate them now, and then set `spec.security.manageNetworkPolicy` to `false` as outlined in the migration checklists.
```
